python/lsst/meas/extensions/shapeHSM/_hsm_shape.py

```diff
@@ -0,0 +1,372 @@
+        schema : `~lsst.afw.table.Schema`
+            The schema to which the field is added.
+        doc : `str`
+            The documentation string that needs to be updated to reflect the
+            type and component of the measurement.
+
+        Returns
+        -------
+        `~lsst.afw.table.KeyD`
+            The key associated with the added field in the schema.
+        """
+        componentLookup = {1: "+ component", 2: "x component"}
+        typeLookup = {"e": " of ellipticity", "g": " of estimated shear"}
+        name = f"{name}_{self.measTypeSymbol}{n}"
+        updatedDoc = f"{doc} ({componentLookup[n]}{typeLookup[self.measTypeSymbol]})"
+        return schema.addField(name, type=float, doc=updatedDoc)
+
+    def measure(self, record, exposure):
+        """
+        Measure adaptive moments of sources given an exposure and set the
+        results in the record in place.
+
+        Parameters
+        ----------
+        record : `~lsst.afw.table.SourceRecord`
+            The record where measurement outputs will be stored.
+        exposure : `~lsst.afw.image.Exposure`
+            The exposure containing the source which needs measurement.
+
+        Raises
+        ------
+        MeasurementError
+            Raised for errors in measurement.
+        """
+        # Extract the centroid from the record.
+        center = self.centroidExtractor(record, self.flagHandler)
+
+        if self.hasDeblendKey and record.get(self.deblendKey) > 0:
+            raise measBase.MeasurementError(self.PARENT_SOURCE.doc, self.PARENT_SOURCE.number)
+
+        # Get the bounding box of the source's footprint.
+        bbox = record.getFootprint().getBBox()
+
+        # Check that the bounding box has non-zero area.
+        if bbox.getArea() == 0:
+            raise measBase.MeasurementError(self.NO_PIXELS.doc, self.NO_PIXELS.number)
+
+        # Ensure that the centroid is within the bounding box.
+        if not bbox.contains(Point2I(center)):
+            raise measBase.MeasurementError(self.NOT_CONTAINED.doc, self.NOT_CONTAINED.number)
+
+        # Get the PSF image evaluated at the source centroid.
+        psfImage = exposure.getPsf().computeImage(center)
+        psfImage.setXY0(0, 0)
+
+        # Get the trace radius of the PSF.
+        psfSigma = exposure.getPsf().computeShape(center).getTraceRadius()
+
+        # Turn bounding box corners into GalSim bounds.
+        bounds = self.bboxToGalSimBounds(bbox)
+
+        # Get the bounding box of the PSF in the parent coordinate system.
+        psfBBox = psfImage.getBBox(afwImage.PARENT)
+
+        # Turn the PSF bounding box corners into GalSim bounds.
+        psfBounds = self.bboxToGalSimBounds(psfBBox)
+
+        # Each GalSim image below will match whatever dtype the input array is.
+        # NOTE: PSF is already restricted to a small image, so no bbox for
+        # the PSF is expected.
+        image = galsim._Image(exposure.image[bbox].array, bounds, None)
```

Comment:
```suggestion
        image = galsim._Image(exposure.image[bbox].array, bounds, wcs=None)
```

and in other places.

Reply:
Done!